packages/turbo/src/BridgeComponent.ts

```diff
@@ -0,0 +1,89 @@
+import { Component } from 'react';
+import type { EmitterSubscription } from 'react-native';
+import type {
+  VisitableViewModule,
+  StradaEvent,
+  StradaMessages,
+  StradaReactComponentProps,
+} from './types';
+import { getNativeModule, registerMessageEventListener } from './common';
+
+const RNVisitableViewModule = getNativeModule<VisitableViewModule>(
+  'RNVisitableViewModule'
+);
+
+class BridgeComponent extends Component {
+  private name: string;
+  private url: string;
+  private sessionHandle: string;
+  private eventHandlerEventSubscription?: EmitterSubscription;
+  private previousMessages: StradaMessages = {};
+
+  constructor(props: StradaReactComponentProps) {
```

Comment:
Considering if we should throw an error if `name` is not defined. This is a required static for components to work.

Reply:
`name` is always defined on a class (it defaults to the class name):

```
> class Test {}
> Test.name
> 'Test'
```

Reply:
Ah, awesome, TIL :-)